src/components/Search/CompareWithBase.tsx

```diff
@@ -159,12 +159,46 @@ function CompareWithBase({
     setInProgressNewRevs(revisionsNew);
   };
 
-  const handleSearchResultsToggleBase = (toggleArray: Changeset[]) => {
-    setInProgressBaseRevs(toggleArray || []);
+  const handleItemToggleInChangesetList = (
+    item: Changeset,
+    maxRevisions: number,
+    changesets: Changeset[],
+  ) => {
+    const isChecked = changesets.map((rev) => rev.id).includes(item.id);
+    const newChecked = [...changesets];
+
+    // if item is not already checked, add to checked
+    if (changesets.length < maxRevisions && !isChecked) {
+      newChecked.push(item);
+    } else if (isChecked) {
+      // if item is already checked, remove from checked
+      newChecked.splice(changesets.indexOf(item), 1);
+    } else {
+      // if there are already `maxRevisions` checked revisions, print a warning
+      const variant: VariantType = 'warning';
+      enqueueSnackbar(`Maximum ${maxRevisions} revision(s).`, { variant });
+    }
+
+    const filteredChecked = [...new Set(newChecked)];
+    return filteredChecked;
+  };
+
+  const handleSearchResultsToggleBase = (item: Changeset) => {
+    const newBaseRevs = handleItemToggleInChangesetList(
+      item,
+      1,
+      baseInProgressRevs,
+    );
+    setInProgressBaseRevs(newBaseRevs);
   };
 
-  const handleSearchResultsToggleNew = (toggleArray: Changeset[]) => {
-    setInProgressNewRevs(toggleArray || []);
+  const handleSearchResultsToggleNew = (item: Changeset) => {
+    const newNewRevs = handleItemToggleInChangesetList(
+      item,
+      3,
+      newInProgressRevs,
+    );
+    setInProgressNewRevs(newNewRevs);
```

Comment:
I had to scroll up to see what are `1` and `3`.

Reply:
very good remark!
you know what, I'll change `handleItemToggleInChangesetList` to take an option object, so that the parameters are clearer

Reply:
I added a commit https://github.com/mozilla/perfcompare/pull/626/commits/1e7894533c5b79f22943c161813c80400986bc0d, tell me what you think!